### DISCUSSION TITLE
Getting Base65 String Instead of Number[] - Backend->Frontend

### DISCUSSION THREAD

**Discussion by KiddoV** [wailsapp/wails#2094] [Q&A]
I have a function defined in Go:
```go
func (a *App) GetFileContent() (fileDat []byte, err error) {
	if filePath, err := runtime.OpenFileDialog(a.ctx, runtime.OpenDialogOptions{Title: "Please choose a file..."}); err != nil {
		return nil, err
	} else {
		fileDat, err = os.ReadFile(filePath)
		return fileDat, err
	}

}
```
In TS I would call the function
```ts
import { GetFileContent } from '../../wailsjs/go/main/App';

const getFileContent = () => {
        GetFileContent().then(file => {
            console.log(file);
        }).catch(err => {
            console.log(err);
        })
}
```

``App.d.ts``
```ts
export function GetFileContent():Promise<Array<number>>;
```
...but my TS function``getFileContent()`` would return base64 (string) in the console.

This make the export function in ``App.d.ts`` unrealizable, since it would return a different type (type ``Promise<string>`` instead of type ``Promise<Array<number>>``) as how it was automatically defined in ``wailsjs`` folder.

How do I resolve this mistype issue?
Thanks!

**Comment by leaanthony** [comment]
The short answer is because Go Marshalls `[]byte` as a base64 encoded string when using JSON. If you're loading a file, a custom AssetHandler might be better?

**Reply by KiddoV** [reply]
I've never done custom AssetHandler. Can you give some code examples?

**Reply by leaanthony** [reply]
https://wails.io/docs/guides/dynamic-assets

**Reply by KiddoV** [reply]
Not what I intended to do, but thanks!
